### ISSUE TITLE
Fix header line generation for CSV data export

### ISSUE THREAD

**Issue by elfring** [traviscross/mtr#451]
I built the evolving network analysis tool according to the development revision “v0.95-17-g826ffa9”.
I tried the program out then accordingly.

```
Sonne:/home/altes_Heim2/elfring/Projekte/Bau/mtr/console # ./mtr --report-cycles 5 --report --csv download.opensuse.org
Mtr_Version,Start_Time,Status,Host,Hop,Ip,Loss%,Snt, ,Last,Avg,Best,Wrst,StDev,
…
```

:thought_balloon: Such a header line should not contain two empty field names (in the middle and at the end).

**Comment by elfring** [comment]
💭
* [The concatenation order can be adjusted for the delimiter](https://github.com/traviscross/mtr/blob/0886eee5bab164fee50f94d542de71fca8a2dc68/ui/report.c#L491 "Update candidate: csv_close()").
* How do you think about to mark any field as “inactive” by default?

**Comment by rewolff** [comment]
I think the default "field list" contains "spacer" so it prints that field in the results. And it puts a header on top. It takes a lot of code to fix that and it is currently "ok" in that an empty field prints with an empty header.

**Comment by elfring** [comment]
>… it is currently "ok" in that an empty field prints with an empty header.

I disagree with is detail because I observed that the column numbers would be different between the header and subsequent record sets.

**Comment by rewolff** [comment]
Make it easy for me to verify that. Paste output here.

**Comment by elfring** [comment]
Do you find a test command (like the following) helpful for further clarification of involved issues? :thinking:

```
Sonne:…/Bau/mtr/console # my_file=/tmp/mtr-test-20220914.csv && ./mtr --report-cycles 2 --csv localhost > ${my_file} && head -n 1 ${my_file} | tr ',' '\n' | wc -l && tail -n 1 ${my_file} | tr ',' '\n' | wc -l
…
```

**Comment by mroe1234** [comment]
I'm going to agree that it is /not/ ok to have empty fields.  the CSV output is useful for programmatic uses and having a literal space as a header 1. doesn't even really make sense to begin with, and 2. can cause weirdness with tools farther down the pipeline.

**Comment by rewolff** [comment]
![mtr-csv](https://user-images.githubusercontent.com/1678121/231490682-ccde6c81-d6c8-4e78-a9a8-582c5f8e9c36.png)

Stuff seems to align correctly. There is just a , at the end of the header line. 

If you provide a patch to remove that , I'll accept it. 
If you provide a patch to remove the empty header and "0" column below it, I'll accept it.

**Comment by elfring** [comment]
How do you think about to replace a field name “space character” by a more meaningful one for the header line? :thinking:

**Comment by rewolff** [comment]
Oh,... Something like "empty column" would better describe what it is, right?

**Comment by elfring** [comment]
:thought_balloon: Can an empty column be omitted finally?

**Comment by rewolff** [comment]
Just so you here know what's going on... 

Darafei Praliaskouski is handling some backlog of issues, and pull requests. 
I just had a discussion with him and I decided that it NOT a good idea to remove the empty columns. People have the option to set the format string to "no spaces" to prevent empty columns from being generated. But others may have "gone with the flow" and created automated workflows that count on the empty column. So... Historically, a space in the format string will give you an empty column, for backwards compatiblity I don't dare remove that. Sorry.
